apps/files_sharing/src/models/Share.js

```diff
@@ -43,6 +43,14 @@ export default class Share {
 		ocsData.hide_download = !!ocsData.hide_download
 		ocsData.mail_send = !!ocsData.mail_send
 
+		if (ocsData.attributes) {
+			try {
+				ocsData.attributes = JSON.parse(ocsData.attributes)
+			} catch (e) {
+				console.warn('Could not parse share attributes returned by server: "' + ocsData.attributes + '"')
+			}
+		}
```

Comment:
As the later parts assume that attributes is an array:

```suggestion
		} else {
			ocsData.attributes = []
		}
```

This makes sure that changing the attributes works properly in the frontend if none are present yet